src/App.jsx

```diff
@@ -94,8 +95,69 @@ export default function App() {
   const [shareMenuOpen, setShareMenuOpen] = useState(false);
   const [successfulCopies, setSuccessfulCopies] = useState([]);
   const [isResetModalOpen, setIsResetModalOpen] = useState(false);
+  const [isLoadMenuOpen, setIsLoadMenuOpen] = useState(false);
 
   const copyPasteRef = useRef();
+  const inputRef = useRef();
+
+  /* https://stackoverflow.com/questions/67399620/how-to-make-open-url-on-click-on-button-in-reactjs */
+  const openInNewTab = (url) => {
+    const newWindow = window.open(url, '_blank', 'noopener,noreferrer');
+    if (newWindow) newWindow.opener = null;
+  };
+
+  /* https://medium.com/@ctrlaltmonique/how-to-create-a-custom-file-upload-button-in-react-with-typescript-b08150f1532e */
+  function handleFileUpload(e) {
+    const { files } = e.target;
+    if (!files) return;
+
+    const file = files[0];
+    const reader = new FileReader();
+    reader.onload = () => {
+      /* process the player.log file */
+      const fileContentArray = reader.result.split('\n');
+      let userString;
+      for (let i = 0; i < fileContentArray.length; i += 1) {
+        const line = fileContentArray[i];
+        if (line.trim().startsWith('===> {"code":0,"user"')) {
+          userString = line;
+          break;
+        }
+      }
+      const userData = JSON.parse(userString.slice(4).trim()).user;
+      const inventory = userData.cards;
+      const divRoot = document.getElementById('decksHolder');
+      const root = createRoot(divRoot);
+      const DECKS = [];
+      userData.decks.forEach((deck) => {
+        const cardIds = [];
+        deck.cards.forEach((card) => {
+          inventory.forEach((invCard) => {
+            // eslint-disable-next-line no-underscore-dangle
+            if (invCard._id === card) {
+              cardIds.push(invCard.cardid);
+            }
+          });
+        });
+        const madeurl = `/?deck=${cardIds.join('.')}`;
+        DECKS.push({ name: deck.name, url: madeurl });
+      });
+      root.render(
+        <div>
+          {DECKS.map((deck) => (
+            <Button label={deck.name} style={{ margin: '8px' }} onClick={() => openInNewTab(deck.url)} />))}
+        </div>,
+      );
```

Comment:
```suggestion
      root.render(
        DECKS.map((deck) => (
          <Button
            label={deck.name}
            style={{ margin: '8px' }}
            onClick={() => openInNewTab(deck.url)}
          />
        )),
      );
```

Remove the surrounding div so the elements are properly flexed. (This works because an array is a valid Reactdom element)

If ever in the future you need something to be a JSX element but don't want to render a parent div, use a `<React.fragment>stuff</React.fragment>` or the shorthand `<>stuff</>`

Reply:
Thanks for letting me know! I'm not very used to using React haha